src/syslog.rs

```diff
@@ -0,0 +1,146 @@
+//! Interfaces for controlling system log.
+
+use crate::NixPath;
+use crate::Result;
+use std::ffi::OsStr;
+
+/// Logging options of subsequent [`syslog`] calls can be set by calling [`openlog`].
+///
+/// The parameter `ident` is a string that will be prepended to every message. The `logopt`
+/// argument specifies logging options. The `facility` parameter encodes a default facility to be
+/// assigned to all messages that do not have an explicit facility encoded.
+pub fn openlog<P: NixPath + ?Sized>(
+    ident: &P,
```

Comment:
On Linux, the `ident` argument is suggested to have the static lifetime: 

> The argument ident in the call of `openlog()` is probably stored as-is.  Thus, if the string it points to is changed, `syslog()` may start prepending the changed string, and if the string it points to ceases to exist, the results are undefined.  Most portable is to use a string constant.

I think we will have a different interface for Linux in the future:

```rs
pub fn openlog(
    ident: &'static Cstr,
```

Reply:
I just realized that using `S: AsRef<OsStr> + ?Sized>` looks exactly right for the `syslog(..., message)` argument, maybe `ident` should also be done similarly, considering they are not paths but strings.